apm-sniffer/apm-agent-core/src/main/java/org/apache/skywalking/apm/agent/core/context/IgnoredTracerContext.java

```diff
@@ -32,30 +32,33 @@
 public class IgnoredTracerContext implements AbstractTracerContext {
     private static final NoopSpan NOOP_SPAN = new NoopSpan();
 
+    private final CorrelationContext correlationContext;
+
     private int stackDepth;
 
     public IgnoredTracerContext() {
         this.stackDepth = 0;
+        this.correlationContext = new CorrelationContext();
     }
 
     @Override
     public void inject(ContextCarrier carrier) {
-
+        carrier.getCorrelationContext().resetFrom(this.correlationContext);
```

Comment:
`inject` is for further propagation, I am confused the implementation here.

Reply:
Add the `inject` and `extract` into correlation context, and delete the `resetFrom` method.